### PR TITLE
Used focus-within to clean up canvas styles

### PR DESCRIPTION
**Pull request by SamGodwin2**
### What is the context of this PR?
Previously a redux store was being used to track focus styles across the different canvas sections, by using focus-within however the functionality is able to remain the same without needing to use a redux store. 

### How to review 
Tests should pass and all components should look and respond in the same way as they do before this PR.

### CODE REVIEW

File: src/components/EditorSurface/CanvasSection.js [hunk 1]
```diff
@@ -1,54 +1,26 @@
...
-export const BasicSection = styled.div`
+const BasicSection = styled.div`
   padding: 2em 2.5em;
   background-color: white;
   position: relative;
   box-shadow: ${shadow};
   margin-bottom: 1em !important;
-`;
-
-const FocusableSection = styled(BasicSection)`
```
Comment (hamishtaplin) [inline src/components/EditorSurface/CanvasSection.js]: I think we still want this distinction between `BasicSection` and a `FocusableSection` as not all sections need to be focusable.

Comment (WickyNilliams) [reply src/components/EditorSurface/CanvasSection.js]: Reinstate but rename `FocusableSection` -> `CanvasSection`

File: src/components/QuestionPageEditor/index.js [hunk 2]
```diff
@@ -26,14 +24,14 @@ import withDeleteOption from "containers/enhancers/withDeleteOption";
 import * as ToastActionCreators from "redux/toast/actions";
 import QuestionPageToolbar from "./QuestionPageToolbar";
 
-const AddAnswerSection = BasicSection.extend`
+const AddAnswerSection = CanvasSection.extend`
```
Comment (hamishtaplin) [inline src/components/QuestionPageEditor/index.js]: This should be a `BasicSection`—we don't want to highlight it when children have focus (hence the need to keep `BasicSection` around).

File: src/components/EditorSurface/CanvasSection.js [hunk 3]
```diff
@@ -21,34 +11,18 @@ export const BasicSection = styled.div`
...
-  transition: outline-color 100ms ease-in;
-  margin-bottom: 1px;
-  outline: 1px solid transparent;
-  ${props => props.isFocused && focusedStyle};
-
+const CanvasSection = styled(BasicSection)`
   &:hover {
+    outline: 1px solid transparent;
     outline-color: ${rgba(colors.blue, 0.5)};
```
Comment (samiwel) [inline src/components/EditorSurface/CanvasSection.js]: Can the `-color` not be collapsed into the line above?

**Pull request merged**